application/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if .Values.networkPolicy }}
+{{- if .Values.networkPolicy.enabled }}
+apiVersion: "networking.k8s.io/v1"
+kind: NetworkPolicy
+metadata:
+  labels:
+{{ include "application.labels.stakater" . | indent 4 }}
+{{ include "application.labels.chart" . | indent 4 }}
+{{- if .Values.networkPolicy.additionalLabels }}
+{{ toYaml .Values.networkPolicy.additionalLabels | indent 4 }}
+{{- end }}
+{{- if or .Values.networkPolicy.annotations }}
+  annotations:
+{{ toYaml .Values.networkPolicy.annotations | indent 4 }}
+{{- end }}
+  name: {{ template "application.name" . }}-network-policy
```

Comment:
we are not following a naming convention where we mention resource type in its name.  you can remove -network-policy